cocotb/share/lib/vhpi/VhpiCbHdl.cpp

```diff
@@ -764,16 +764,38 @@ VhpiStartupCbHdl::VhpiStartupCbHdl(GpiImplInterface *impl) : GpiCbHdl(impl),
     cb_data.reason = vhpiCbStartOfSimulation;
 }
 
-int VhpiStartupCbHdl::run_callback(void) {
+int VhpiStartupCbHdl::run_callback() {
+    vhpiHandleT tool, argv_iter, argv_hdl;
     gpi_sim_info_t sim_info;
-    sim_info.argc = 0;
-    sim_info.argv = NULL;
-    sim_info.product = gpi_copy_name(vhpi_get_str(vhpiNameP, NULL));
-    sim_info.version = gpi_copy_name(vhpi_get_str(vhpiToolVersionP, NULL));
-    gpi_embed_init(&sim_info);
+    char **tool_argv = NULL;
+    uint32_t tool_argc = 0;
+    int i = 0;
+
+    tool = vhpi_handle(vhpiTool, NULL);
+    
+    sim_info.product = (char *)vhpi_get_str(vhpiNameP, tool);
+    sim_info.version = (char *)vhpi_get_str(vhpiToolVersionP, tool);
```

Comment:
Since this is C++, could you switch to C++-style `static_cast` here and below?

Reply:
I updated it using `static_cast` and `const_cast`.